components/org.wso2.carbon.identity.api.server.branding.preference.management/org.wso2.carbon.identity.api.server.branding.preference.management.v1/src/main/java/org/wso2/carbon/identity/api/server/branding/preference/management/v1/core/BrandingPreferenceManagementService.java

```diff
@@ -83,37 +70,35 @@
     public BrandingPreferenceModel addBrandingPreference(BrandingPreferenceModel brandingPreferenceModel) {
 
         String tenantDomain = getTenantDomainFromContext();
-        /**
-         * Currently this API provides the support to only configure tenant wise branding preference for 'en-US' locale.
-         * So always use resource name as default resource name.
-         */
-        String resourceName = getDefaultResourceName();
-        // Check whether a branding resource already exists with the same name in the particular tenant to be added.
-        if (isResourceExists(BRANDING_RESOURCE_TYPE, resourceName)) {
-            throw handleException(Response.Status.CONFLICT, ERROR_CODE_CONFLICT_BRANDING_PREFERENCE, tenantDomain);
-        }
         String preferencesJSON = generatePreferencesJSONFromRequest(brandingPreferenceModel.getPreference());
         if (!BrandingPreferenceUtils.isValidJSONString(preferencesJSON)) {
             throw handleException(Response.Status.BAD_REQUEST, ERROR_CODE_INVALID_BRANDING_PREFERENCE, null);
         }
 
+        BrandingPreference requestDTO, responseDTO;
```

Comment:
It feels little bit odd to use the same object for request and response.

Reply:
We were decided to have same object for request and response when we deciding the API defenition. Do we need to change it?